### PR TITLE
Toggle selected cells

### PR DESCRIPTION
**Pull request by wcandillon**

### CODE REVIEW

File: app/directive/layoutmodel.js [hunk 1]
```diff
@@ -48,6 +48,10 @@ angular.module('layoutmodel', [ 'lodash', 'ui.bootstrap' ])
                     return $sce.trustAsHtml(accounting.formatNumber(fact.Value, decimals));
                 };
 
+                scope.cellClick = function(data) {
+                    data.clicked = !data.clicked;
```
Comment (dknochen) [inline app/directive/layoutmodel.js]: could you rename "clicked" to "selected"

File: app/directive/layoutmodel.html [hunk 2]
```diff
@@ -78,7 +78,7 @@
         </td>
 
         <!-- data cells -->
-        <td class="data" ng-click="dataclick({ data : data })" ng-class="classes(data, headerGroup[headerGroup.length - 1])" ng-repeat="data in data[$index] track by $index" ng-include="dataTemplate">
+        <td class="data" ng-click="dataclick({ data : data }); cellClick(data)" ng-class="[{ 'selected-cell': data.clicked }, classes(data, headerGroup[headerGroup.length - 1])]" ng-repeat="data in data[$index] track by $index" ng-include="dataTemplate">
```
Comment (dknochen) [inline app/directive/layoutmodel.html]: I'm not convinced if we really need two separate functions... but its ok.

**Comment by dknochen** [comment]
@wcandillon feel free to merge after addressing the comments...

File: app/directive/layoutmodel.js [hunk 3]
```diff
@@ -48,19 +48,27 @@ angular.module('layoutmodel', [ 'lodash', 'ui.bootstrap' ])
...
                 };
 
+                scope.cellClick = function(data) {
+                    if(_.isObject(data)) {
+                        data.clicked = !data.clicked;
+                    }
+                };
+
                 scope.classes = function(data, header) {
                     /*jshint eqnull:true */
```
Comment (wcandillon) [inline app/directive/layoutmodel.js]: @dknochen This function was really not ready for populate=false

File: app/directive/layoutmodel.js [hunk 4]
```diff
@@ -48,19 +48,27 @@ angular.module('layoutmodel', [ 'lodash', 'ui.bootstrap' ])
                     return $sce.trustAsHtml(accounting.formatNumber(fact.Value, decimals));
                 };
 
+                scope.cellClick = function(data) {
+                    if(_.isObject(data)) {
+                        data.clicked = !data.clicked;
```
Comment (dknochen) [inline app/directive/layoutmodel.js]: could you rename "clicked" to "selected"

File: app/directive/layoutmodel.html [hunk 5]
```diff
@@ -78,7 +78,7 @@
         </td>
 
         <!-- data cells -->
-        <td class="data" ng-click="dataclick({ data : data })" ng-class="classes(data, headerGroup[headerGroup.length - 1])" ng-repeat="data in data[$index] track by $index" ng-include="dataTemplate">
+        <td class="data" ng-click="dataclick({ data : data }); selectCell(data)" ng-class="[{ 'selected-cell': data.selected }, classes(data, headerGroup[headerGroup.length - 1])]" ng-repeat="data in data[$index] track by $index" ng-include="dataTemplate">
```
Comment (wcandillon) [inline app/directive/layoutmodel.html]: @dknochen Indeed we might do something more generic than that and remove the call to `selectCell`

**Pull request merged**